src/persist-client/src/cfg.rs

```diff
@@ -0,0 +1,278 @@
+    /// compaction requests that can execute for a given shard.
+    pub compaction_concurrency_limit: usize,
+    /// In Compactor::compact_and_apply_background, the maximum number of pending
+    /// compaction requests to queue.
+    pub compaction_queue_size: usize,
+    /// The maximum number of concurrent blob deletes during garbage collection.
+    pub gc_batch_part_delete_concurrency_limit: usize,
+    /// In Compactor::compact_and_apply_background, the minimum amount of time to
+    /// allow a compaction request to run before timing it out. A request may be
+    /// given a timeout greater than this value depending on the inputs' size
+    pub compaction_minimum_timeout: Duration,
+    /// The maximum size of the connection pool to Postgres/CRDB when performing
+    /// consensus reads and writes.
+    pub consensus_connection_pool_max_size: usize,
+    /// The minimum TTL of a connection to Postgres/CRDB before it is proactively
+    /// terminated. Connections are routinely culled to balance load against the
+    /// downstream database.
+    pub consensus_connection_pool_ttl: Duration,
+    /// The minimum time between TTLing connections to Postgres/CRDB. This delay is
+    /// used to stagger reconnections to avoid stampedes and high tail latencies.
+    /// This value should be much less than `consensus_connection_pool_ttl` so that
+    /// reconnections are biased towards terminating the oldest connections first.
+    /// A value of `consensus_connection_pool_ttl / consensus_connection_pool_max_size`
+    /// is likely a good place to start so that all connections are rotated when the
+    /// pool is fully used.
+    pub consensus_connection_pool_ttl_stagger: Duration,
+    /// The # of diffs to initially scan when fetching the latest consensus state, to
+    /// determine which requests go down the fast vs slow path. Should be large enough
+    /// to fetch all live diffs in the steady-state, and small enough to query Consensus
+    /// at high volume. Steady-state usage should accommodate readers that require
+    /// seqno-holds for reasonable amounts of time, which to start we say is 10s of minutes.
+    ///
+    /// This value ought to be defined in terms of `NEED_ROLLUP_THRESHOLD` to approximate
+    /// when we expect rollups to be written and therefore when old states will be truncated
+    /// by GC.
+    pub state_versions_recent_live_diffs_limit: usize,
+    /// Length of time after a writer's last operation after which the writer
+    /// may be expired.
+    pub writer_lease_duration: Duration,
+    /// Length of time after a reader's last operation after which the reader
+    /// may be expired.
+    pub reader_lease_duration: Duration,
+    /// Length of time between critical handles' calls to downgrade since
+    pub critical_downgrade_interval: Duration,
+    /// Hostname of this persist user. Stored in state and used for debugging.
+    pub hostname: String,
+}
+
+// Tuning inputs:
```

Comment:
Nit: if might be better placed over the `struct` definition item, so they are also rendered in RustDoc, especially if the selected values are not meant to be hard-coded in the long-term. If those are going to be exposed as `SystemVars` fields, I suggest to put the tuning rationale in the `SystemVar` description and replicate this text in the `LaunchDarkly` description.

Reply:
putting it as rustdoc on the struct is a great idea, done! I also like the idea of moving them into SystemVar+LaunchDarkly descriptions, but that'll have to wait until we actually have SystemVars for most of these, so left that one as a TODO